### DISCUSSION TITLE
Recompile not being triggered by changing source file

### DISCUSSION THREAD

**Discussion by tballard** [com-lihaoyi/mill#3234] [Development]
As requested, I've been trying to modify a mill shared code example enough that it fails like my real project. Of course it works perfectly. In the real code, the problem occurs in shared code, (at least) so I have put a simple shared class in each, with a method returning a String, which is used in a "print" statement on both client and server.

Doing a clean and compile works fine for both.

```
mill path compile share.jvm
```
is the same for both.

Changing the shared file and recompiling works as you would hope for the toy. For the real project, not so much.

```
find . -name '*\.log' | xargs ls -latr
```
shows 3 changed files for the toy
```
-rw-rw-r--  1 tballard  staff   186 Jun 25 07:41 ./out/share/js/compile.log
-rw-rw-r--  1 tballard  staff  1093 Jun 25 07:41 ./out/client/fastLinkJS.log
-rw-rw-r--  1 tballard  staff   187 Jun 25 07:41 ./out/share/jvm/compile.log
```
and none for the real project.

I'm hoping there is some good way to see the difference in what is happening in the 2 cases. What do you recommend?

By the way, "visualize" and "visualizePlan" are failing with with a GrapvizException saying "None of the provided engines could be initialized".

**Comment by lefou** [comment]
Can you provide a reproducer and more infos about your environment?
